simplecpp.cpp

```diff
@@ -2187,14 +2187,19 @@ namespace simplecpp {
                 continue;
             }
             // get previous subpath
-            const std::string::size_type pos1 = path.rfind('/', pos - 1U) + 1U;
-            const std::string previousSubPath = path.substr(pos1, pos-pos1);
+            std::string::size_type pos1 = path.rfind('/', pos - 1U);
+            if (pos1 == std::string::npos) {
+                pos1 = 0;
+            } else {
+                pos1 += 1U;
+            }
+            const std::string previousSubPath = path.substr(pos1, pos - pos1);
```

Comment:
Is it possible to add a test case? Also, somewhat related, would it make sense to add sanitizer tests to simplecpp CI as well?

Reply:
> Is it possible to add a test case?

Not sure if I could figure one out - there's one in Cppcheck which triggers it. I just tried to fix the code and keep the test intact :D

> Also, somewhat related, would it make sense to add sanitizer tests to simplecpp CI as well?

Was wondering that myself...

Reply:
Probably cppcheck performs more tests than this project itself.
But adding 2-3 additional CI runs might help anyway, see PR #197 "valgrind"